agent/consul/state/config_entry_test.go

```diff
@@ -310,6 +409,440 @@ func TestStore_ConfigEntries(t *testing.T) {
+	require.Len(t, gatewayServices, 1)
+	require.Equal(t, gatewayServices[0].Kind, structs.GatewayservicekindDestination)
+
+	require.NoError(t, s.EnsureNode(0, &structs.Node{Node: "node1"}))
+	require.NoError(t, s.EnsureService(0, "node1", service))
+
+	//Watch is fired because we transitioned to a destination, by default we assume it's not.
+	require.True(t, watchFired(ws))
+
+	_, gatewayServices, err = s.GatewayServices(ws, "Gtwy1", nil)
+	require.NoError(t, err)
+	require.Len(t, gatewayServices, 1)
+	require.Equal(t, gatewayServices[0].Kind, structs.GatewayservicekindService)
+
+	ws = memdb.NewWatchSet()
+	_, _, err = s.GatewayServices(ws, "Gtwy1", nil)
+	require.NoError(t, err)
+
+	require.NoError(t, s.DeleteService(6, "node1", service.ID, &service.EnterpriseMeta, ""))
+
+	//Watch is fired because we transitioned to a destination, by default we assume it's not.
+	require.True(t, watchFired(ws))
+
+	_, gatewayServices, err = s.GatewayServices(ws, "Gtwy1", nil)
+	require.NoError(t, err)
+	require.Len(t, gatewayServices, 1)
+	require.Equal(t, gatewayServices[0].Kind, structs.GatewayservicekindDestination)
+
+}
+
+func TestStore_Service_TerminatingGateway_Kind_Service(t *testing.T) {
+	s := testConfigStateStore(t)
+
+	Gtwy := &structs.TerminatingGatewayConfigEntry{
+		Kind: structs.TerminatingGateway,
+		Name: "Gtwy1",
+		Services: []structs.LinkedService{
+			{
+				Name: "web",
+			},
+		},
+	}
+
+	// Create
+	require.NoError(t, s.EnsureConfigEntry(0, Gtwy))
+
+	service := &structs.NodeService{
+		Kind:    structs.ServiceKindTypical,
+		Service: "web",
+	}
+
+	_, gatewayServices, err := s.GatewayServices(nil, "Gtwy1", nil)
+	require.NoError(t, err)
+	require.Len(t, gatewayServices, 1)
+	require.Equal(t, gatewayServices[0].Kind, structs.GatewayservicekindUnknown)
+
+	ws := memdb.NewWatchSet()
+	_, _, err = s.GatewayServices(ws, "Gtwy1", nil)
+	require.NoError(t, err)
+
+	// Create
+	require.NoError(t, s.EnsureNode(0, &structs.Node{Node: "node1"}))
+	require.NoError(t, s.EnsureService(0, "node1", service))
+
+	//Watch is fired because we transitioned to a destination, by default we assume it's not.
+	require.True(t, watchFired(ws))
+
+	_, gatewayServices, err = s.GatewayServices(ws, "Gtwy1", nil)
+	require.NoError(t, err)
+	require.Len(t, gatewayServices, 1)
+	require.Equal(t, gatewayServices[0].Kind, structs.GatewayservicekindService)
+
+	ws = memdb.NewWatchSet()
+	_, _, err = s.GatewayServices(ws, "Gtwy1", nil)
+	require.NoError(t, err)
+
+	require.NoError(t, s.DeleteService(6, "node1", service.ID, &service.EnterpriseMeta, ""))
+
+	//Watch is fired because we transitioned to a destination, by default we assume it's not.
+	require.True(t, watchFired(ws))
+
+	_, gatewayServices, err = s.GatewayServices(ws, "Gtwy1", nil)
+	require.NoError(t, err)
+	require.Len(t, gatewayServices, 1)
+	require.Equal(t, gatewayServices[0].Kind, structs.GatewayservicekindUnknown)
+
+}
+
+func TestStore_ServiceDefaults_Kind_Destination_Wildcard(t *testing.T) {
```

Comment:
Same comment here about redundancy with `TestStore_ServiceDefaults_isDestination_Wildcard`

Reply:
right all the `isDestination` tests are duplicate. I will remove them